File: 2739-total-distance-traveled/2739-total-distance-traveled.cpp
// class Solution {
// public:
//     int distanceTraveled(int mainTank, int additionalTank) {
//         int distance=0;
//         int total_fuel=0;
//         while(mainTank>=5){
//             mainTank=mainTank+(mainTank-5);
//             if(additionalTank>0){
//                 mainTank++;
//                 additionalTank--;
//             }
//         }
//         total_fuel+=mainTank;
//         return total_fuel;
//     }
// };

//TC-> O(1)
// SC-> O(1)
class Solution {
public:
     int distanceTraveled(int mainTank, int additionalTank) {
        return (mainTank+min((mainTank-1)/4,additionalTank))*10;
    }
};

File: 2739-total-distance-traveled/2739-total-distance-traveled.java
// class Solution {
//      public static int distanceTraveled(int m, int a) {
//         int d=0,tf=0;

//         while(m>=5){
//             tf+=5;
//             m=m-5;
//             if(a>0){
//                 m++;
//                 a--;
//             }
//         }
//         tf+=m;
//         return tf*10;
//     }

// }

// TC-> O(1)
// SC-> O(1)
class Solution {
     public static int distanceTraveled(int m, int a) {
         
         return (m+Math.min((m-1)/4,a))*10;
}
}